cpp/src/parquet/arrow/reader_writer_benchmark.cc

```diff
@@ -118,13 +134,22 @@ std::shared_ptr<::arrow::Table> TableFromVector(
 
 template <>
 std::shared_ptr<::arrow::Table> TableFromVector<BooleanType>(const std::vector<bool>& vec,
-                                                             bool nullable) {
+                                                             bool nullable,
+                                                             int null_percentage) {
   BooleanBuilder builder;
   if (nullable) {
     std::vector<bool> valid_bytes(BENCHMARK_SIZE, 0);
-    int n = {0};
-    std::generate(valid_bytes.begin(), valid_bytes.end(),
-                  [&n] { return (n++ % 2) != 0; });
+    if (null_percentage == -1) {
```

Comment:
Perhaps nulls generation can be factored out?

Reply:
done.